sdk/core/azure-core/src/main/java/com/azure/core/implementation/AccessControllerUtils.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.implementation;
+
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+
+/**
+ * Utility class for {@code AccessController} APIs.
+ * <p>
+ * This class exists to warden off usages of {@code java.security} APIs into a separate class that if it has loading
+ * issues usages of {@code AccessController} APIs will be skipped.
+ */
+public final class AccessControllerUtils {
+    private static final boolean CAN_BE_USED;
+
+    static {
+        ClassLoader classLoader = AccessControllerUtils.class.getClassLoader();
```

Comment:
Is there any possibility the static block in this class could be called after somebody tries to call `doPrivileged()` or `doPriviliegedException()`? I vaguely remember us having to call an empty constructor on classes that use `JsonSerializable` because of some weird race condition, but I might be mistaken.

Reply:
Nope, the static constructor will be called during class load, so it must happen before the static APIs are called.